apps/blockchain/lib/blockchain/interface/account_interface.ex

```diff
@@ -185,7 +208,26 @@ defimpl EVM.Interface.AccountInterface, for: Blockchain.Interface.AccountInterfa
           {:ok, integer()} | :account_not_found | :key_not_found
   def get_storage(account_interface, evm_address, key) do
     address = Account.Address.from(evm_address)
-    Account.get_storage(account_interface.state, address, key)
+    cached_value = Cache.get_current_value(account_interface.cache, address, key)
+
+    case cached_value do
+      nil -> Account.get_storage(account_interface.state, address, key)
+      :deleted -> :key_not_found
```

Comment:
We're returning `:key_not_found` to match the behavior of the `Account.get_storage` when a key isn't found, right? If so, I wonder if there's a better way to consolidate the knowledge of `:key_not_found` somewhere else? For example, maybe the `Account` module itself makes use of the cache if it finds one?